static/js/components/ShareDataForm.jsx

```diff
@@ -0,0 +1,224 @@
+      setSelectedPhotRows([]);
+      setSelectedSpecRows([]);
+    }
+    setIsSubmitting(false);
+  };
+
+  if (!photometry[route.id] || !allGroups || !spectra[route.id]) {
+    return <>Loading...</>;
+  }
+
+  const photRows = photometry[route.id].map((phot) => (
+    createPhotRow(phot.id, phot.mjd, phot.mag, phot.magerr, phot.limiting_mag,
+      phot.instrument_name, phot.filter)));
+
+  const specRows = spectra[route.id].map((spec) => (
+    createSpecRow(spec.id, spec.instrument_name)));
+
+  const options = {
+    textLabels: {
+      body: {
+        noMatch: "",
+      }
+    },
+    filter: true,
+    selectableRows: "multiple",
+    selectableRowsOnClick: true,
+    filterType: "dropdown",
+    responsive: "vertical",
+    rowsPerPage: 10,
+    selectableRowsHeader: true,
+    customToolbarSelect: () => { },
+    download: false,
+    print: false
+  };
+
+  return (
+    <>
+      <div>
+        <Typography variant="h5">
+          Share Source Data
+        </Typography>
+      </div>
+      <br />
+      <div>
+        <MUIDataTable
+          columns={photHeadCells}
+          data={photRows}
+          title="Photometry"
+          options={{
+            ...options,
+            rowsSelected: selectedPhotRows,
+            onRowSelectionChange: (rowsSelectedData, allRows, rowsSelected) => {
+              setSelectedPhotRows(rowsSelected);
+            }
+          }}
+        />
+        <br />
+        <MUIDataTable
+          columns={specHeadCells}
+          data={specRows}
+          title="Spectra"
+          options={{
+            ...options,
+            rowsSelected: selectedSpecRows,
+            onRowSelectionChange: (rowsSelectedData, allRows, rowsSelected) => {
+              setSelectedSpecRows(rowsSelected);
+            }
+          }}
+        />
+      </div>
+      <br />
+      <div>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          {
+            !!errors.groups &&
+              <FormValidationError message="Please select at least one group/user" />
+          }
+          <Controller
+            name="groups"
+            as={(
+              <Autocomplete
+                multiple
+                options={allGroups}
+                getOptionLabel={(group) => group.name}
+                filterSelectedOptions
+                renderInput={(params) => (
+                  <TextField
+                    // eslint-disable-next-line react/jsx-props-no-spreading
+                    {...params}
+                    error={!!errors.groups}
+                    variant="outlined"
+                    label="Select Groups/Users"
+                    className={classes.groupSelect}
+                  />
+                )}
+              />
+              )}
+            control={control}
+            onChange={([, data]) => data}
+            rules={{ validate: validateGroups }}
+            defaultValue={[]}
+          />
+          <br />
+          <div>
+            <Button
```

Comment:
Currently there is no error raised if one tries to share a spectrum or photometry point with someone it is already shared with. Is that expected behavior?

Reply:
Yes